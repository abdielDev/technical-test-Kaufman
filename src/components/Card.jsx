import React from 'react';
import '../css/components/Card.css';

const Card = ({ handleClick, title, price, image }) => {
  return (
    <div onClick={handleClick} className="Card">
      <img src={image} alt={title}/>
      <h2>{title}</h2>
      <p>{`$${price}` || 'Omitted'}</p>
      <button onClick={handleClick}>Purchase</button>
    </div>
  );
};

export default Card;