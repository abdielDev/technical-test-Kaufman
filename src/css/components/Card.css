.Card {
  display: block;
  padding: 10px;
  margin-bottom: 20px;
  overflow: hidden;
  width: auto;
  background-color: var(--color-white-primary);
  text-align: center;
  color: var(--color-black-primary);
}

.Card img {
  margin-bottom: 10px;
}

.Card h2 {
  margin-bottom: 10px;
}

.Card p {
  margin-bottom: 10px;
}

.Card em {
  float: right;
  font-style: italic;
}

.Card button {
  padding: 10px;
  background-color: var(--color-button-primary);
  border: none;
  cursor: pointer;
  font-weight: bold;
  color: var(--color-white-primary);
}