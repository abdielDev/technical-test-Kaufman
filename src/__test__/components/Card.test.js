import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import mockProduct from '../../__mocks__/productMock';
import Card from '../../components/Card';

describe('<Card />', () => {
  const card = shallow(<Card title={mockProduct.title} price={mockProduct.price} image={mockProduct.thumbnail} />);
  test('Component Card render', () => {
    expect(card.length).toEqual(1);
  });

  test('It received the props', () => {
    expect(card.find('h2').text()).toBe("");
  });
});

describe('Card Snapshot', () => {
  test('Check UI of component Card', () => {
    const card = create(<Card title={mockProduct.title} price={mockProduct.price} image={mockProduct.thumbnail} />);
    expect(card.toJSON()).toMatchSnapshot();
  });
});