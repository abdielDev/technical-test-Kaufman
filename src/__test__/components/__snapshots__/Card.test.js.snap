// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Snapshot Check UI of component Card 1`] = `
<div
  className="Card"
>
  <img
    alt=""
  />
  <h2>
    
  </h2>
  <p>
    $null
  </p>
  <button>
    Purchase
  </button>
</div>
`;
